src/lib/rank_utils.ts

```diff
@@ -22,9 +22,74 @@ interface IRankInfo {
     label: string;
 }
 
-export const MaxRank: number = 36;
+class Rating {
+    unset:boolean;
+    rating:number;
+    deviation:number;
+    volatility:number;
+    rank:number;
+    rank_label:string;
+    partial_rank:number;
+    partial_rank_label:string;
+    rank_deviation_labels:Array<string>;
+    rank_deviation:number;
+}
+
+export const MaxRank: number = 39;
+
+
+const MIN_RATING = 100;
+const MAX_RATING = 6000;
+
+export function rank_to_rating(rank:number) {
+    return 900 * Math.exp(0.032 * rank);
```

Comment:
The EGF system uses a variable `a` depending on your rating to make it harder to progress the higher you go. Traditional Elo doesn't have a variable `a`, but also doesn't have to worry about handicaps, so basically what the EGF has done is to slow down movement in the Dan ranks so as to keep their model of `100 points = 1 rank` consistent. 

That same kind of system isn't going to work for glicko though, the whole point of glicko is to be able to hone in on a player's appropriate rating quicker, so dampening their progression kinda defeats the point and fundamentally fights the system.

So, instead the approach I took is to make each rank take more points to earn, that way we're not mucking with the way glicko2 works, but we're still achieving the same goal of making it harder to progress through the Dan ranks.

FYI the `900 * exp(0.032 * rank)` formula was derived by computing everyone's new glicko ratings, mapping them to what their current rank is under the EGF system, and fitting a curve to that data. The model definitely falls apart at the lower ddk ranks, anywhere `<20k` or so is pretty much nonsense, but that was true with the EGF system too, and in general kinda makes sense, in that range there's a ton of variability in games and trying to figure out good handicap values to use is pretty much impossible. We will most likely stop doing handicaps by default for low DDK's because of this.